src/Microsoft.AspNet.Mvc.Core/ModelBinders/BodyModelBinder.cs

```diff
@@ -19,39 +17,25 @@ namespace Microsoft.AspNet.Mvc.ModelBinding
     /// </summary>
     public class BodyModelBinder : BindingSourceModelBinder
     {
-        private readonly ActionContext _actionContext;
-        private readonly IScopedInstance<ActionBindingContext> _bindingContext;
-        private readonly IInputFormatterSelector _formatterSelector;
-        private readonly IValidationExcludeFiltersProvider _bodyValidationExcludeFiltersProvider;
-
         /// <summary>
         /// Creates a new <see cref="BodyModelBinder"/>.
         /// </summary>
-        /// <param name="context">An accessor to the <see cref="ActionContext"/>.</param>
-        /// <param name="bindingContext">An accessor to the <see cref="ActionBindingContext"/>.</param>
-        /// <param name="selector">The <see cref="IInputFormatterSelector"/>.</param>
-        /// <param name="bodyValidationExcludeFiltersProvider">
-        /// The <see cref="IValidationExcludeFiltersProvider"/>.
-        /// </param>
-        public BodyModelBinder([NotNull] IScopedInstance<ActionContext> context,
-                               [NotNull] IScopedInstance<ActionBindingContext> bindingContext,
-                               [NotNull] IInputFormatterSelector selector,
-                               [NotNull] IValidationExcludeFiltersProvider bodyValidationExcludeFiltersProvider)
+        public BodyModelBinder()
             : base(BindingSource.Body)
         {
-            _actionContext = context.Value;
-            _bindingContext = bindingContext;
-            _formatterSelector = selector;
-            _bodyValidationExcludeFiltersProvider = bodyValidationExcludeFiltersProvider;
         }
 
         /// <inheritdoc />
         protected async override Task<ModelBindingResult> BindModelCoreAsync([NotNull] ModelBindingContext bindingContext)
         {
-            var formatters = _bindingContext.Value.InputFormatters;
+            var requestServices = bindingContext.OperationBindingContext.HttpContext.RequestServices;
+
+            var formatterSelector = requestServices.GetRequiredService<IInputFormatterSelector>();
```

Comment:
Started a thread - https://github.com/aspnet/Mvc/issues/2330 - to discuss if we could change the contract for formatters. It might allow us to get away from doing `GetRequiredService` here.